### DISCUSSION TITLE
Updating repository through API disables auto-merge

### DISCUSSION THREAD

**Discussion by alejandrohdezma** [community/community#24301] [Apps, API and Webhooks]
<p>I noticed that when a repository has the “Allow auto-merge” option enabled, performing a PATCH request to update any of the other settings in the repository results in said option being disabled.</p>
<p>For example:</p>
<pre><code class="lang-auto">curl -X PATCH \
  -H "Authorization: token $GITHUB_TOKEN" \
  -d '{"has_wiki": false}' \
  https://api.github.com/repos/owner/repo
</code></pre>
<p>Given that the auto-merge option cannot be updated through the API, this problem makes it impossible to use the API to synchronize the settings of several repositories, since each time this is done, you would have to manually update the auto-merge setting.</p>
<p>Is this behaviour expected? Or is it a bug?</p>

**Comment by canuckjacq** [comment]
<p>:wave: Welcome!</p>
<p>No, that is not expected behaviour. Thank you for the report. We’ve raised an issue and hopefully they’ll get that sorted soon!</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>Hi <a class="mention" href="/alejandrohdezma">@alejandrohdezma</a>! We pushed a fix for this yesterday. Please let us know if you’re still having trouble.</p>

**Comment by alejandrohdezma** [comment]
<p>Working like a charm now! Thank you very much <a class="mention" href="/canuckjacq">@canuckjacq</a>! That was quick!</p>
